clar2wasm/src/wasm_generator.rs

```diff
@@ -247,6 +247,27 @@ impl WasmGenerator {
         })
     }
 
+    pub fn expand_memory_conditionally(&mut self) -> Result<(), GeneratorError> {
+        let memory = self
+            .module
+            .memories
+            .iter_mut()
+            .next()
+            .ok_or_else(|| GeneratorError::InternalError("No Memory found".to_owned()))?;
+
+        let total_module_space = memory.initial * 64 * 1024;
+        let new_memory_required = self.literal_memory_end;
+
+        if new_memory_required > total_module_space {
+            let pages_required = (new_memory_required - total_module_space) / (64 * 1024);
+            let remainder = (new_memory_required - total_module_space) % (64 * 1024);
+
+            memory.initial += pages_required + (remainder > 0) as u32;
+        }
```

Comment:
This is still expanding instead of setting.
I would just assign `memory.initial` to the end value, like

```suggestion
        memory.initial = self.literal_memory_end >> 16 + (self.literal_memory_end & 0xffff != 0) as u32;
```
(please don't commit this suggestion, it's just me having fun, and it's horrible)

Reply:
I'm not sure If I'm following you correctly. So something like this?

```rust
        let mut end_memory = memory.initial;

        if new_memory_required > total_module_space {
            let pages_required = (new_memory_required - total_module_space) / (64 * 1024);
            let remainder = (new_memory_required - total_module_space) % (64 * 1024);

            end_memory += pages_required + (remainder > 0) as u32;
        }

        memory.initial = end_memory;
```

Reply:
No, what I mean is that we have by default 10 pages. What your current code does is that it will grow those pages if we need more than 10 pages.
But I think we should just set the number of pages that we need.
So if you compute that the required number of pages is 4, you should set 4.

Reply:
Ah right. Got it.